app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/attachment/AttachmentOptions.kt

```diff
@@ -9,27 +9,39 @@ import androidx.compose.foundation.layout.PaddingValues
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.lazy.GridCells
 import androidx.compose.foundation.lazy.LazyVerticalGrid
+import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import com.wire.android.R
 import com.wire.android.appLogger
 import com.wire.android.ui.common.AttachmentButton
 import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.home.conversations.model.AttachmentBundle
+import com.wire.android.ui.home.messagecomposer.AttachmentInnerState
+import com.wire.android.ui.home.messagecomposer.AttachmentState
 import com.wire.android.util.permission.UseCameraRequestFlow
+import com.wire.android.util.permission.UseStorageRequestFlow
 import com.wire.android.util.permission.rememberCaptureVideoFlow
 import com.wire.android.util.permission.rememberCurrentLocationFlow
 import com.wire.android.util.permission.rememberOpenFileBrowserFlow
 import com.wire.android.util.permission.rememberOpenGalleryFlow
 import com.wire.android.util.permission.rememberRecordAudioRequestFlow
 import com.wire.android.util.permission.rememberTakePictureFlow
 
-@OptIn(ExperimentalComposeUiApi::class, ExperimentalFoundationApi::class)
+@OptIn(ExperimentalComposeUiApi::class, ExperimentalFoundationApi::class, ExperimentalMaterial3Api::class)
 @Composable
-fun AttachmentOptionsComponent() {
-    val attachmentOptions = buildAttachmentOptionItems()
+fun AttachmentOptionsComponent(
+    attachmentInnerState: AttachmentInnerState,
+    onSendAttachment: (AttachmentBundle?) -> Unit,
+    onError: (String) -> Unit
+) {
+    val attachmentOptions = buildAttachmentOptionItems(attachmentInnerState)
```

Comment:
what if we move this line inside LazyVerticalGrid ? on the place where we are iterating over attachmentOptions as we are not using val attachmentOptions anywhere I do not think we evern need this local variable

Reply:
I'll leave it as it is, because sadly I think it's not possible since this `buildAttachmentOptionItems` function needs to be created inside a composable context

Reply:
ahhh makes sense, sure no worries